src/cow-react/api/gasPrices/index.ts

```diff
@@ -0,0 +1,122 @@
+import { DEFAULT_NETWORK_FOR_LISTS } from 'constants/lists'
+import { SupportedChainId as ChainId } from '@cowprotocol/cow-sdk'
+import { GAS_FEE_ENDPOINTS, GAS_API_KEYS } from 'constants/index'
+
+// Values are returned as floats in gwei
+const ONE_GWEI = 1_000_000_000
+
+export type GasFeeEndpointResponse = {
+  lastUpdate: string
+  average: string | null
+  fast: string | null
+  slow: string | null
+}
+
+export interface EstimatedPrice {
+  confidence: number
+  price: number
+  maxFeePerGas: number
+  maxPriorityFeePerGas: number
+}
+
+type GasPrices = Omit<GasFeeEndpointResponse, 'lastUpdate'>
+
+// Here the key is what we use in gas state and the value is confidence level
+// coming from the Blockscout api, so we map state values with confidence lvls
+const priceMap: GasPrices = {
+  fast: '99',
+  average: '90',
+  slow: '70',
+}
+
+class GasFeeApi {
```

Comment:
I'm fine, but being purist the API responsibility is not right. 

We should have different API implementations, for different APIs. 
Then we just compose them to make our own logic. 

This way we would for example, create BlockNativeApi or sth like that, and then separate the API  from our business logic

Reply:
It's hard to know what's our business logic if we mix it with API implementation. Moreover, is harder to add backup gas price API, replace them, etc.